tests/test_data_handling.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+import requests
+
+from fobi_custom.plugins.form_handlers.collect_data.fobi_form_handlers import CollectDataPlugin
+from fobi.dynamic import assemble_form_class
+from fobi.models import FormEntry
+
+from pldp.models import Survey, SurveyRow, SurveyComponent
+
+@pytest.mark.django_db
+def test_data_handler(sample_form_entry, sample_location, sample_study, sample_form_element, mocker):
+    request = mocker.MagicMock(spec=requests.Response)
+    form = assemble_form_class(form_entry=sample_form_entry)
+
+    # form.cleaned_data['c75e27cf-4d8f-49dc-bb15-da8137dac247'] = 5
```

Comment:
All right. This is tricky. I looked at [`django-fobi` for inspiration.](https://github.com/barseghyanartur/django-fobi/blob/070ee3239cc4df3f5e841ee8649d8c26a10f007e/src/fobi/tests/test_dynamic_forms.py#L35)

```
# Instantiate an instance of the `assemble_form_class`.
form_class = assemble_form_class(form_entry=sample_form_entry)

# Pass data to it
form = form_class(data={'c75e27cf-4d8f-49dc-bb15-da8137dac247': 5})

assert form.is_valid() == True
```

I have a couple questions about this:
* I do not understand the GUIDs as form fields. Do you know what fobi has going on here?
* What is your motivation for creating a `DynamicForm` with data? Your `run` function does not appear to do anything with it directly, right?

Reply:
- Fobi has a 'name' field on each form element, which it uses to connect saved submitted data with other form data (eg plugin_uid, position, etc.). It's not the way I would have set up the package, necessarily, but changing it would mean doing some pretty major surgery to Fobi's base functions. It's essentially an additional ID for each form element, as I understand it.
- The run function saves submitted data! It should end up in the `saved_data` column of SurveyComponent. See: https://github.com/datamade/just-spaces/blob/data-handler/fobi_custom/plugins/form_handlers/collect_data/fobi_form_handlers.py#L67